diffrax/integrate.py

```diff
@@ -836,13 +1197,21 @@ def _promote(yi):
             raise ValueError(
                 "`max_steps=None` is incompatible with `saveat.dense=True`"
             )
-        (
-            _,
-            _,
-            dense_info,
-            _,
-            _,
-        ) = solver.step(terms, tprev, tnext, y0, args, solver_state, made_jump)
+        else:
+            (
+                _,
+                _,
+                dense_info,
+                _,
+                _,
+            ) = solver.step(terms_, tprev, tnext, y0, args, solver_state, made_jump)
+            if delays is not None:
+                new_discontinuities = jnp.full(delays.max_discontinuities, jnp.inf)
+                new_discontinuities = new_discontinuities.at[
+                    jnp.arange(len(delays.initial_discontinuities))
+                ].set(delays.initial_discontinuities)
+                discontinuities_save_index = len(delays.initial_discontinuities)
```

Comment:
I think this can be done more efficiently via
```python
buffer = jnp.full(delays.max_discontinuities - len(delays.initial_discontinuities), jnp.inf)
new_discontinuities = jnp.concatenate([delays.initial_discontinuities, buffer])
```